videointelligence/annotate/logo_detection.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package annotate contains speech examples.
+package annotate
+
+// [START video_detect_logo]
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	video "cloud.google.com/go/videointelligence/apiv1"
+	"github.com/golang/protobuf/ptypes"
+	videopb "google.golang.org/genproto/googleapis/cloud/videointelligence/v1"
+)
+
+// logoDetection analyzes a video and extracts logos with their bounding boxes.
+func logoDetection(w io.Writer, filename string) error {
+	// filename := "resources/googlework_tiny.mp4"
+
+	ctx := context.Background()
+
+	// Creates a client.
+	client, err := video.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("video.NewClient: %v", err)
+	}
+
+	fileBytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return fmt.Errorf("ioutil.ReadFile: %v", err)
+	}
+
+	op, err := client.AnnotateVideo(ctx, &videopb.AnnotateVideoRequest{
+		InputContent: fileBytes,
+		Features: []videopb.Feature{
+			videopb.Feature_LOGO_RECOGNITION,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("AnnotateVideo: %v", err)
+	}
+
+	resp, err := op.Wait(ctx)
+	if err != nil {
+		return fmt.Errorf("Wait: %v", err)
+	}
+
+	// Only one video was processed, so get the first result.
+	result := resp.GetAnnotationResults()[0]
+
+	// Annotations for list of logos detected, tracked and recognized in video.
+	for _, annotation := range result.LogoRecognitionAnnotations {
+		fmt.Fprintf(w, "Description: %q\n", annotation.Entity.GetDescription())
+		// Opaque entity ID. Some IDs may be available in [Google Knowledge
+		// Graph Search API](https://developers.google.com/knowledge-graph/).
```

Comment:
Markdown links unfortunately don't work.
```suggestion
		// Opaque entity ID. Some IDs may be available in Google Knowledge
		// Graph Search API (https://developers.google.com/knowledge-graph/).
```